lib/src/sql/array.rs

```diff
@@ -317,13 +318,18 @@ pub trait Uniq<T> {
 
 impl Uniq<Array> for Array {
 	fn uniq(mut self) -> Array {
-		for x in (0..self.len()).rev() {
-			for y in (x + 1..self.len()).rev() {
-				if self[x] == self[y] {
-					self.remove(y);
-				}
+		let mut set: HashSet<&Value> = HashSet::new();
+		let mut to_remove: Vec<usize> = Vec::new();
+		for i in 0..self.len() {
+			if set.contains(&self[i]) {
+				to_remove.push(i);
+			} else {
+				set.insert(&self[i]);
```

Comment:
Instead of `contains` followed by `insert`, can unconditionally insert and then use the return value of `insert` to figure out if it was already present. This could halve the number of map lookups.